modules/websocket/websocket-grammar.ym

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (c) 2014 BalaBit IT Ltd, Budapest, Hungary
+ * Copyright (c) 2016 Yilin Li <liyilin1214@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+%code requires {
+
+#include "websocket-parser.h"
+
+}
+
+%code {
+
+#include "cfg-parser.h"
+#include "cfg-grammar.h"
+#include "plugin.h"
+
+}
+
+%name-prefix "websocket_"
+%lex-param {CfgLexer *lexer}
+%parse-param {CfgLexer *lexer}
+%parse-param {LogDriver **instance}
+%parse-param {gpointer arg}
+
+
+/* INCLUDE_DECLS */
+
+%token KW_WEBSOCKET
+%token KW_MODE
+%token KW_ADDRESS
+%token KW_PORT
+%token KW_PROTOCOL
+%token KW_PATH
+%token KW_SSL
+%token KW_CERT
+%token KW_KEY
+%token KW_CACERT
+%token KW_ALLOW_SELF_SIGNED
+%token KW_TEMPLATE
+%%
+
+start
+        : LL_CONTEXT_DESTINATION KW_WEBSOCKET
+          {
+            last_driver = *instance = websocket_dd_new(configuration);
```

Comment:
It  is really a wired problem. It really takes me a while to find the bug that causes it ...... 
It's because I use #ifndef WEBSOCKET_H_INCLUDED in both websocket-destination.h  and websocket-source.h 
Fixed now :)